lib/at_host/at_host.c

```diff
@@ -53,28 +52,62 @@ enum select_uart {
 
 static enum term_modes term_mode;
 static struct device *uart_dev;
-static int at_socket_fd = INVALID_DESCRIPTOR;
-static struct pollfd fds[1];
-static int nfds;
 static u8_t at_buf[AT_MAX_CMD_LEN];
 static size_t at_buf_len;
-static struct k_work at_cmd_send_work;
-static struct k_thread socket_thread;
-static K_THREAD_STACK_DEFINE(socket_thread_stack, THREAD_STACK_SIZE);
+static struct k_work cmd_send_work;
+static const char termination[3] = { '\0', '\r', '\n' };
 
 
-static const char termination[3] = { '\0', '\r', '\n' };
 
-static void at_cmd_send(struct k_work *work)
+static inline void write_uart_string(char *str, size_t len)
 {
-	int bytes_sent;
+	for (size_t i = 0; i < len; i++) {
+		uart_poll_out(uart_dev, str[i]);
+	}
+}
+
+static void response_handler(char *response)
+{
+	int len = strlen(response) + 1;
+	/* Forward the data over UART */
+	if (len > 0) {
+		write_uart_string(response, len);
+	}
+}
+
+static void cmd_send(struct k_work *work)
+{
+	size_t            chars;
+	char              str[15];
+	char              buf[AT_MAX_CMD_LEN] = {0};
+	enum at_cmd_state state;
+	int               err;
 
 	ARG_UNUSED(work);
 
-	bytes_sent = send(at_socket_fd, at_buf, at_buf_len, 0);
+	err = at_cmd_write(at_buf, buf, AT_MAX_CMD_LEN, &state);
+	if (err != 0) {
+		LOG_ERR("Could not send AT command to modem: %d", err);
```

Comment:
If you cannot send the AT command to the Modem, I think you should not print an error on the UART but you should return here?

Reply:
Fixed